### PR TITLE
Fix handling of negative offsets for time w/ tz

### PR DESCRIPTION
**Pull request by martint**

### CODE REVIEW

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/encoder/json/TestCustomJsonDateTimeFormatter.java [hunk 1]
```diff
@@ -84,12 +84,12 @@ public void testCustomDateTimeFunctions()
...
-        testTimeWithTZ(sqlTimeWithTimeZoneOf(3, 10, 23, 35, 123000000, 2, 0), "HH:mm:ss.SSS Z", "10:23:35.123 +0200");
-        testTimeWithTZ(sqlTimeWithTimeZoneOf(3, 10, 23, 35, 123000000, 10, 0), "HH:mm:ss.SSS Z", "10:23:35.123 +1000");
-        testTimeWithTZ(sqlTimeWithTimeZoneOf(3, 10, 23, 35, 123000000, -10, 0), "HH:mm:ss.SSS Z", "10:23:35.123 -1000");
-        testTimeWithTZ(sqlTimeWithTimeZoneOf(3, 23, 59, 59, 999000000, 0, 0), "HH:mm:ss.SSS Z", "23:59:59.999 +0000");
+        testTimeWithTZ(sqlTimeWithTimeZoneOf(3, 10, 23, 35, 123000000, 0), "HH:mm:ss.SSS Z", "10:23:35.123 +0000");
+        testTimeWithTZ(sqlTimeWithTimeZoneOf(3, 0, 0, 0, 0, 0), "HH:mm:ss.SSS Z", "00:00:00.000 +0000");
+        testTimeWithTZ(sqlTimeWithTimeZoneOf(3, 10, 23, 35, 123000000, 2 * 60), "HH:mm:ss.SSS Z", "10:23:35.123 +0200");
+        testTimeWithTZ(sqlTimeWithTimeZoneOf(3, 10, 23, 35, 123000000, 10 * 60), "HH:mm:ss.SSS Z", "10:23:35.123 +1000");
+        testTimeWithTZ(sqlTimeWithTimeZoneOf(3, 10, 23, 35, 123000000, -10 * 60), "HH:mm:ss.SSS Z", "10:23:35.123 -1000");
+        testTimeWithTZ(sqlTimeWithTimeZoneOf(3, 23, 59, 59, 999000000, 0), "HH:mm:ss.SSS Z", "23:59:59.999 +0000");
```
Comment (findepi) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/encoder/json/TestCustomJsonDateTimeFormatter.java:92]: it's preexisting, but we just learned it's so important

please add a test with minutes offsets here (positive, negative, -00:XX)

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/encoder/json/TestISO8601JsonDateTimeFormatter.java [hunk 2]
```diff
@@ -84,12 +84,12 @@ public void testISO8601DateTimeFunctions()
...
-        testTimeWithTZ(sqlTimeWithTimeZoneOf(3, 10, 23, 35, 123000000, 2, 0), "10:23:35.123+02:00");
-        testTimeWithTZ(sqlTimeWithTimeZoneOf(3, 10, 23, 35, 123000000, 10, 0), "10:23:35.123+10:00");
-        testTimeWithTZ(sqlTimeWithTimeZoneOf(3, 10, 23, 35, 123000000, -10, 0), "10:23:35.123-10:00");
-        testTimeWithTZ(sqlTimeWithTimeZoneOf(3, 23, 59, 59, 999000000, 0, 0), "23:59:59.999Z");
+        testTimeWithTZ(sqlTimeWithTimeZoneOf(3, 10, 23, 35, 123000000, 0), "10:23:35.123Z");
+        testTimeWithTZ(sqlTimeWithTimeZoneOf(3, 0, 0, 0, 0, 0), "00:00:00Z");
+        testTimeWithTZ(sqlTimeWithTimeZoneOf(3, 10, 23, 35, 123000000, 2 * 60), "10:23:35.123+02:00");
+        testTimeWithTZ(sqlTimeWithTimeZoneOf(3, 10, 23, 35, 123000000, 10 * 60), "10:23:35.123+10:00");
+        testTimeWithTZ(sqlTimeWithTimeZoneOf(3, 10, 23, 35, 123000000, -10 * 60), "10:23:35.123-10:00");
+        testTimeWithTZ(sqlTimeWithTimeZoneOf(3, 23, 59, 59, 999000000, 0), "23:59:59.999Z");
```
Comment (findepi) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/encoder/json/TestISO8601JsonDateTimeFormatter.java:92]: it's preexisting, but we just learned it's so important

please add a test with minutes offsets here (positive, negative, -00:XX)

File: presto-main/src/test/java/io/prestosql/operator/scalar/timestamptz/TestTimestampWithTimeZone.java [hunk 3]
```diff
@@ -1279,6 +1279,21 @@ public void testCastToTimeWithTimeZone()
...
+        assertThat(assertions.expression("CAST(TIMESTAMP '2020-03-08 01:23:45.123456 -08:35' AS TIME(6) WITH TIME ZONE)")).matches("TIME '01:23:45.123456-08:35'");
+        assertThat(assertions.expression("CAST(TIMESTAMP '2020-03-08 01:23:45.1234567 -08:35' AS TIME(7) WITH TIME ZONE)")).matches("TIME '01:23:45.1234567-08:35'");
+        assertThat(assertions.expression("CAST(TIMESTAMP '2020-03-08 01:23:45.12345678 -08:35' AS TIME(8) WITH TIME ZONE)")).matches("TIME '01:23:45.12345678-08:35'");
+        assertThat(assertions.expression("CAST(TIMESTAMP '2020-03-08 01:23:45.123456789 -08:35' AS TIME(9) WITH TIME ZONE)")).matches("TIME '01:23:45.123456789-08:35'");
+        assertThat(assertions.expression("CAST(TIMESTAMP '2020-03-08 01:23:45.1234567891 -08:35' AS TIME(10) WITH TIME ZONE)")).matches("TIME '01:23:45.1234567891-08:35'");
+        assertThat(assertions.expression("CAST(TIMESTAMP '2020-03-08 01:23:45.12345678912 -08:35' AS TIME(11) WITH TIME ZONE)")).matches("TIME '01:23:45.12345678912-08:35'");
+        assertThat(assertions.expression("CAST(TIMESTAMP '2020-03-08 01:23:45.123456789123 -08:35' AS TIME(12) WITH TIME ZONE)")).matches("TIME '01:23:45.123456789123-08:35'");
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/operator/scalar/timestamptz/TestTimestampWithTimeZone.java:1296]: add -00:35 cases here too

File: presto-main/src/test/java/io/prestosql/operator/scalar/timetz/TestTimeWithTimeZone.java [hunk 4]
```diff
@@ -1110,6 +1217,21 @@ public void testCastToTimestampWithTimeZone()
...
+        assertThat(assertions.expression("CAST(TIME '12:34:56.1234567-08:35' AS TIMESTAMP(7) WITH TIME ZONE)", session)).matches("TIMESTAMP '2020-05-01 12:34:56.1234567 -08:35'");
+        assertThat(assertions.expression("CAST(TIME '12:34:56.12345678-08:35' AS TIMESTAMP(8) WITH TIME ZONE)", session)).matches("TIMESTAMP '2020-05-01 12:34:56.12345678 -08:35'");
+        assertThat(assertions.expression("CAST(TIME '12:34:56.123456789-08:35' AS TIMESTAMP(9) WITH TIME ZONE)", session)).matches("TIMESTAMP '2020-05-01 12:34:56.123456789 -08:35'");
+        assertThat(assertions.expression("CAST(TIME '12:34:56.1234567891-08:35' AS TIMESTAMP(10) WITH TIME ZONE)", session)).matches("TIMESTAMP '2020-05-01 12:34:56.1234567891 -08:35'");
+        assertThat(assertions.expression("CAST(TIME '12:34:56.12345678912-08:35' AS TIMESTAMP(11) WITH TIME ZONE)", session)).matches("TIMESTAMP '2020-05-01 12:34:56.12345678912 -08:35'");
+        assertThat(assertions.expression("CAST(TIME '12:34:56.123456789123-08:35' AS TIMESTAMP(12) WITH TIME ZONE)", session)).matches("TIMESTAMP '2020-05-01 12:34:56.123456789123 -08:35'");
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/operator/scalar/timetz/TestTimeWithTimeZone.java:1233]: negative offset minutes case?

File: presto-spi/src/main/java/io/prestosql/spi/type/SqlTimeWithTimeZone.java [hunk 5]
```diff
@@ -91,7 +92,7 @@ public String toString()
             builder.append(".");
             builder.append(format("%0" + precision + "d", scaledFraction));
         }
-        builder.append(format("%+03d:%02d", offsetMinutes / 60, offsetMinutes % 60));
+        builder.append(format("%+03d:%02d", offsetMinutes / 60, abs(offsetMinutes % 60)));
```
Comment (findepi) [inline presto-spi/src/main/java/io/prestosql/spi/type/SqlTimeWithTimeZone.java:95]: cast to varchar too:

```
presto> SELECT CAST(TIME '01:23:45-08:35' AS varchar);
      _col0
-----------------
 01:23:45-08:-35
(1 row)
```

**Review by findepi — APPROVED** [review]
LGTM; one outstanding correctness and a bit more consistent testing cover
